argo_rollouts/tests/test_e2e.py

```diff
@@ -0,0 +1,21 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .common import E2E_METRICS
+
+
+@pytest.mark.e2e
+def test_e2e_openmetrics_v2(dd_agent_check):
+    aggregator = dd_agent_check(rate=True)
+
+    aggregator.assert_service_check('argo_rollouts.openmetrics.health', ServiceCheck.OK, count=2)
+    for metric in E2E_METRICS:
+        aggregator.assert_metric(metric)
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```

Comment:
nit - we now have a function to assert the service checks defines in the config file as well: https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/utils.py#L152